hyperspy/misc/utils.py

```
@@ -1543,3 +1546,89 @@ def is_binned(signal, axis=-1):
         return signal.metadata.Signal.binned
     else:
         return signal.axes_manager[axis].is_binned
+
+
+def is_cupy_array(array):
+    """
+    Convenience function to determine if an array is a cupy array
+
+    Parameters
+    ----------
+    array : array
+        The array to determine whether it is a cupy array or not.
+
+    Returns
+    -------
+    bool
+        True if it is cupy array, False otherwise.
+
+    """
+    try:
+        import cupy as cp
+        return isinstance(array, cp.ndarray)
+    except ImportError:
+        return False
+
+
+def to_numpy(array):
```

Just from the name and first line of the docstring one might expect this to work on any kind of array, also to compute dask arrays. I'm not sure whether there is a better alternative. As a general comment to these functions, I'm wondering whether the dask array case should be considered here or not.

---

Good point, shall we raise an error in case of dask array?

---

Yes I think an error should be fine, might catch accidental incorrect use of to_numpy elsewhere.